### ISSUE TITLE
Version 3.8.0 - Fatal error: Allowed memory size of 134217728 bytes exhausted 

### ISSUE THREAD

**Issue by MircoBabin** [FriendsOfPHP/PHP-CS-Fixer#6357]
## Bug report

OS: Windows/10

PHP CS Fixer 3.8.0 BerSzcz against war! by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.1.4 (64bits version, having no php.ini)

```
php-cs-fixer fix . -vvv >output.txt 2>&1
```

Aborts with:

```
Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 4194312 bytes) in C:\Users\Mirco\AppData\Roaming\Composer\vendor\friendsofphp\php-cs-fixer\src\Fixer\ClassNotation\OrderedClassElementsFixer.php on line 532
```

See also [output.txt](https://github.com/FriendsOfPHP/PHP-CS-Fixer/files/8441091/output.txt).

## On having no php.ini
**Notice: I don't have a php.ini when running php-cs-fixer.**
I attached the output of "php -i" (the php used by php-cs-fixer.bat) in [phpversion.txt](https://github.com/FriendsOfPHP/PHP-CS-Fixer/files/8441107/phpversion.txt). By default the PHP memory limit is 128M.

The project I'm working on is a PHP 5.6.40 project. So when running php.exe the 5.6.40 is used because that is in the _path_. 5.6.40 is not compatible with php-cs-fixer 3.8.0. To get around this limitation I adjusted the %appdata%\Composer\vendor\bin (global Composer) file php-cs-fixer.bat to:

```
@ECHO OFF
setlocal DISABLEDELAYEDEXPANSION
SET BIN_TARGET=%~dp0/php-cs-fixer
SET COMPOSER_RUNTIME_BIN_DIR=%~dp0
set PHPRC=
d:\projects\bin\php-8.1.4-x64\php.exe "%BIN_TARGET%" %*
```

## Expectation

I would expect php-cs-fixer to set memory limit to unlimited. Because it is a CLI application. I see no point in aborting on a 64bits developer machine which has 16 GB of physical memory and around 10 GB free.

## Possible solution
Adjust https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/aa805a8a4d8384c71a1367fceb57a081a4d969e4/php-cs-fixer#L15 with:

```php
ini_set("memory_limit", -1);
```

**Comment by MircoBabin** [comment]
@keradus
1) I don't know what you mean with "SCA tool"?
2) I don't know what you mean with "mess up with PHP settings of local env to for execution"?
If this refers to my proposed "ini_set()" statement, this statement is only an in process memory setting, not persisted on disk, not persisting to other scripts. Not messing up anything, only allowing php-cs-fixer to use as much memory as it needs.

**Comment by keradus** [comment]
ad 1 - Static Code Analysis tools, eg phpstan, psalm, phpcs. or even other, like composer, PHPUnit, etc.
ad 2 - i know it's not persistent, but why should the tool change it on their own? imagine you explicitly wanted to limit the memory to 2gb, and then tool escalate the limit to 10 gb on their own.

**Comment by MircoBabin** [comment]
@keradus 
You do have a point if you explicitly want to limit the memory usage of php-cs-fixer, my proposal would not allow that any more.

I have come up with an alternative solution. Set unlimited memory in php_cs.dist.php

```php

**Comment by MircoBabin** [comment]
@keradus
You do have a point if you explicitly want to limit the memory usage of php-cs-fixer, my proposal would not allow that any more.

I have come up with an alternative solution. Set unlimited memory in .php_cs.dist.php. **This alternative solution works, so I'm closing this issue.** I manually monitored the php memory usage and the peak was around 230 MB. Php-cs-fixer reported "Fixed all files in 20136.852 seconds, 182.000 MB memory used".

The odd thing is that I have to explicitly set the config file on the commandline. I expected it to be used automatically. But maybe that's just my lack of knowledge.

```
cd /d d:\projects\webpage\the-project
rem this directory contains .php_cs.dist.php file

php-cs-fixer fix . --config d:\projects\webpage\the-project\.php_cs.dist.php
```

```php
<?php
// .php_cs.dist.php file

ini_set("memory_limit", -1);

$finder = PhpCsFixer\Finder::create()
    ->in(__DIR__);

$config = new PhpCsFixer\Config();
$config->setUsingCache(false)
    ->setRiskyAllowed(false)
    ->setRules([
        '@PSR2' => true,
        '@Symfony' => true,
    ])
    ->setFinder($finder)
;

return $config;    
```

**Comment by keradus** [comment]
`php-cs-fixer fix -vvv` should pick up the config and show used config on verbose output

you mean you run it for 6h? sth really going wrong there. how big is your codebase? is it public, by any chance?

**Comment by keradus** [comment]
ad 1)

you use config file-name for PHP CS Fixer v2, but using v3 of the tool, rename config file to `.php-cs-fixer.dist.php` or point the filename as argument for `fix` command

ad 2)

ok, 250k files? so its slow, no further questions ;)

**Comment by MircoBabin** [comment]
ad 1) 

Thanx! After creating the .php-cs-fixer.dist.php file it is picked up automatically. 

I saw a message from php-cs-fixer when I first ran it for version 3.8 complaining about .php_cs.dist file was an old file. I did not read the message correctly, I think, and just added the .php suffix resulting in .php_cs.dist.php. As the message was gone I thought everything was correct, not realizing that the filename and config-syntax has changed between v2 and v3.

It all works now. Thank you for your help!
